src/App.jsx

```diff
@@ -1,12 +1,71 @@
 import React, { Component } from "react";
 import Menu from "./components/Menu";
+import LoginForm from "./components/LoginForm";
+import { authenticate } from "./modules/auth";
 
 class App extends Component {
+  state = {
+    renderLoginForm: false,
+    authenticated: false,
+    message: "",
+  };
+
+  onLogin = async (e) => {
+    e.preventDefault();
+    const response = await authenticate(
+      e.target.email.value,
+      e.target.password.value
+    );
+    if (response.authenticated) {
+      this.setState({ authenticated: true });
+    } else {
+      this.setState({ message: response.message, renderLoginForm: false });
+    }
+  };
+
+  onChangeHandler = (e) => {
+    this.setState({ [e.target.name]: e.target.value });
+  };
+
   render() {
+    const { renderLoginForm, authenticated, message } = this.state;
+    let renderLogin;
+    switch (true) {
```

Comment:
Can you try to extract this to a component called `Login`. 
We want to keep the App component as clean as possible.

Reply:
We will move this part into another component when we are adding navbar, we would prefer not to do refactoring twice